client/v2/client.go

```diff
@@ -405,29 +410,100 @@ func NewPointFrom(pt models.Point) *Point {
 	return &Point{pt: pt}
 }
 
+// makeChunks splits blob string into chunks that size
+// of each chunk does not exceed given chunkSize
+// makeChunks assumes that size of blob string is greater than chunkSize
+// makeChunks splits blob by selecting chunk by range
+func makeChunks(blob string, chunkSize int) []string {
+	var (
+		size   = int(math.Ceil(float64(len(blob)) / float64(chunkSize)))
+		chunks = make([]string, 0, size)
+
+		start, stop int
+	)
+
+	// range by chunkSize
+	for i := 0; i < len(blob); i += chunkSize {
+		// set boundaries of range
+		stop = start + chunkSize
+
+		// if the stop is more than blob length
+		// then set it to end of the blob
+		if stop > len(blob) {
+			stop = len(blob)
+		}
+
+		// move deviding position back while do not match a comma char to keep persistency
+		for l := stop; l > 0 && l != len(blob); l-- {
+			stop = l
+			if string(blob[l]) == commaString {
+				break
+			}
+		}
+
+		// if the current position is a comma
+		// then move starting position forward to skip comma sign
+		if string(blob[start]) == commaString {
+			start += 1
+		}
+
+		chunks = append(chunks, blob[start:stop])
+		start = stop
+	}
+	return chunks
+}
+
 func (uc *udpclient) Write(bp BatchPoints) error {
-	var b bytes.Buffer
 	var d time.Duration
 	d, _ = time.ParseDuration("1" + bp.Precision())
 
 	for _, p := range bp.Points() {
-		pointstring := p.pt.RoundedString(d) + "\n"
-
+		pointstring := p.pt.RoundedString(d) + newlineString
 		// Write and reset the buffer if we reach the max size
-		if b.Len()+len(pointstring) >= uc.payloadSize {
-			if _, err := uc.conn.Write(b.Bytes()); err != nil {
-				return err
+		if len(pointstring) >= uc.payloadSize {
+			// If byte size of metrics are more than max UDP payload size(MaxUDPPayload)
+			// then split the metrics into chunks, which each chunk have max MaxUDPPayload byte size
+			var (
+				key, data, timestamp string
+				payload              = strings.Split(pointstring, " ")
```

Comment:
This will break if there are any escape sequences in the "key" of the metric. You could use the pt.Key() function to get the "key", strip it off the front, then use `strings.Rsplit` to split the timestamp off the end.